sync/src/types.rs

```diff
@@ -280,11 +280,90 @@ impl From<(BlockNumber, Byte32)> for BlockNumberAndHash {
     }
 }
 
+enum SchedulerOutput {
+    DoubleInCrease,
+    InCrease,
+    Decrease,
+    DoubleDeCrease,
+}
+
+/// Using 512 blocks as a period, dynamically adjust the scheduler's time standard
```

Comment:
This structure looks like a typical quantiles query algorithm on streaming data,  I'm curious about why [0, 0.5, 0.8, 0.9] quantiles are using here, is it better than [0, 0.25, 0.5, 0.75] ?

Reply:
Yes, the original plan was actually using 0.5 as the boundary, but the results were very poor and would result in a much slower speed. There are several reasons for this.

The purpose of the scheduler is to filter out the relatively good nodes and maintain communication with them. 

If 0.5 is used as a boundary, it may not be a big problem at first, but at a certain point, there will be self-competition, resulting in the number of tasks per node not reaching the desired maximum, which affects the overall efficiency.

Similarly, sampling at 512 intervals is also the result of comparative experiments. If it's too small, it's calculated too often. If it is too large, the purpose of the screening cannot be achieved in the default configuration. Guaranteed number of tasks for nearly 4 nodes with maximum task*4